test/model/tools/model_test/main.cpp

```diff
@@ -244,6 +249,146 @@ load_workload(const char *file)
+ */
+static void
+reduce_counterexample(std::shared_ptr<model::kv_workload> workload, const std::string &home,
+  const std::string &conn_config, const std::string &table_config)
+{
+    /* Separate the workload into sequences. */
+    std::unordered_map<model::txn_id_t, model::kv_workload_sequence_ptr> txn_to_sequence;
+    std::deque<model::kv_workload_sequence_ptr> sequences;
+
+    for (size_t i = 0; i < workload->size(); i++) {
+        model::kv_workload_operation &op = (*workload)[i];
+
+        /* Transaction operations, such as begin, insert, etc. */
+        if (model::operation::transactional(op.operation)) {
+            model::txn_id_t txn_id = model::operation::transaction(op.operation);
+
+            if (std::holds_alternative<model::operation::begin_transaction>(op.operation)) {
+                /* A new transaction. */
+                if (txn_to_sequence.find(txn_id) != txn_to_sequence.end())
+                    throw model::model_exception(
+                      "Transaction ID already taken: " + std::to_string(txn_id));
+
+                model::kv_workload_sequence_ptr seq = std::make_shared<model::kv_workload_sequence>(
+                  sequences.size(), model::kv_workload_sequence_type::transaction);
+                *seq.get() << op.operation;
+                op.seq_no = seq->seq_no();
+                sequences.push_back(seq);
+                txn_to_sequence[txn_id] = seq;
+            } else {
+                /* Existing transactions. */
+                auto itr = txn_to_sequence.find(txn_id);
+                if (itr == txn_to_sequence.end())
+                    throw model::model_exception(
+                      "Transaction ID does not exist: " + std::to_string(txn_id));
+                model::kv_workload_sequence_ptr seq = itr->second;
+                *seq.get() << op.operation;
+                op.seq_no = seq->seq_no();
+
+                /* Transaction end. */
+                if (std::holds_alternative<model::operation::commit_transaction>(op.operation) ||
+                  std::holds_alternative<model::operation::rollback_transaction>(op.operation))
+                    txn_to_sequence.erase(itr);
+            }
+        } else {
+            /* Non-transaction operations, such as set stable timestamp or crash. */
+            model::kv_workload_sequence_ptr seq =
+              std::make_shared<model::kv_workload_sequence>(sequences.size());
+            *seq.get() << op.operation;
+            op.seq_no = seq->seq_no();
+            sequences.push_back(seq);
+
+            /* Operations that clear the transaction state. */
+            if (std::holds_alternative<model::operation::crash>(op.operation) ||
+              std::holds_alternative<model::operation::restart>(op.operation))
+                txn_to_sequence.clear();
+        }
+    }
+
+    /*
+     * Find the minimal subset that causes a failure by using an algorithm similar too binary
```

Comment:
Nitpick: `too binary` -> `to binary`.

Reply:
Fixed in [99d0adb](https://github.com/wiredtiger/wiredtiger/pull/10318/commits/99d0adb57880cb11dd452ea30ba81ee3649e1e9b). Thanks for reading the comments thoroughly :)